packages/core/saved-objects/core-saved-objects-server-internal/src/saved_objects_service.ts

```diff
@@ -122,7 +124,15 @@ export class SavedObjectsService
     const savedObjectsMigrationConfig = await firstValueFrom(
       this.coreContext.configService.atPath<SavedObjectsMigrationConfigType>('migrations')
     );
-    this.config = new SavedObjectConfig(savedObjectsConfig, savedObjectsMigrationConfig);
+    this.config = new SavedObjectConfig(
+      savedObjectsConfig,
+      savedObjectsMigrationConfig,
+      this.allowHttpApiAccess
```

Comment:
I need to inject the value once read from `env` and don't want to pass in the env mode as is.